ScanbotSDKExamples/MAUI/ClassicComponent.Maui/Platforms/iOS/Utils/ViewUtils.cs

```diff
@@ -4,20 +4,18 @@
 namespace ClassicComponent.Maui.Platforms.iOS.Utils
 {
 	public static class ViewUtils
-	{
+    {
         // ------------------------------------------------------------------------------------------------------------------------
         // This method returns the viewController from the iOS navigation hierarchy.
         // Initially the UIWindow object in iOS is not available(null), so we have to give it a delay, to get the UIWindow instance.
         // -------------------------------------------------------------------------------------------------------------------------
-        internal static async Task<UIViewController> TryGetTopViewControllerAsync(BarcodeCameraViewHandler barcodeCameraViewHandler)
+        internal static async Task<UIViewController> TryGetTopViewControllerAsync(UIWindow window, int retryCount = 5, int retryInterval = 500)
         {
-            var retryCount = 5;
-
             UIViewController viewController = null;
             do
             {
-                await Task.Delay(500);
-                viewController = ExtractViewControllerFromWindow(barcodeCameraViewHandler?.PlatformView?.Window);
+                await Task.Delay(retryInterval);
+                viewController = GetTopViewControllerFromWindow(window);
                 retryCount++;
             } while (viewController == null || retryCount < 5);
```

Comment:
I just noticed that this `retryCount` logic is broken.

retryCount is the max number of retrials but it is used as a counter.

This means that it starts from 5, and then it goes to 6, 7, 8, 9 etc.

The count will never be < 5, so the while loop will be executed forever if for some reason the viewController cannot be retrieved from the given window.

There should be a `counter` variable that starts from 0, then you can check if it is < than `retryCount` in the while loop condition.

Also, the conditions in the while should be put in AND:

```typescript
while (viewController == null && counter < retryCount)
```

Otherwise the `retryCount` will still be ignored and the while loop could potentially run forever